src/org/wikibase/data/Time.java

```diff
@@ -0,0 +1,143 @@
+/**
+*  This program is free software; you can redistribute it and/or
+*  modify it under the terms of the GNU General Public License
+*  as published by the Free Software Foundation; either version 3
+*  of the License, or (at your option) any later version. Additionally
+*  this file is subject to the "Classpath" exception.
+*
+*  This program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with this program; if not, write to the Free Software Foundation,
+*  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+package org.wikibase.data;
+
+import java.math.BigInteger;
+import java.net.URL;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+public class Time extends WikibaseDataType {
+    private Calendar calendar;
+    private int before;
+    private int after;
+    private int precision;
+    private URL calendarModel;
+    private BigInteger year;
+
+    public Calendar getCalendar() {
+        return calendar;
+    }
+
+    public void setCalendar(Calendar calendar) {
+        this.calendar = calendar;
+    }
+
+    public int getBefore() {
+        return before;
+    }
+
+    public void setBefore(int before) {
+        this.before = before;
+    }
+
+    public int getAfter() {
+        return after;
+    }
+
+    public void setAfter(int after) {
+        this.after = after;
+    }
+
+    public URL getCalendarModel() {
+        return calendarModel;
+    }
+
+    public void setCalendarModel(URL calendarModel) {
+        this.calendarModel = calendarModel;
+    }
+
+    public int getPrecision() {
+        return precision;
+    }
+
+    public void setPrecision(int precision) {
+        this.precision = precision;
+    }
+
+    public BigInteger getYear() {
```

Comment:
I think we can live with long. I've already run into examples where int isn't good enough so I used BigInteger just to be sure.